### ISSUE TITLE
Training Error: Negative Dimension Size

### ISSUE THREAD

**Issue by thealchemist89** [yhenon/keras-frcnn#139]
Hi everyone,

I am trying to extract features from conv3_x layers. To do this, I commented off Stage 4 conv_block in the function **nn_base()** under **resnet.py**.

Next, to continue the flow of ResNet50 architecture, under the function **classifier_layers()**, I added the conv4_x and conv5_x layer.

However, I experience an output error:

**"ValueError: Negative dimension size caused by subtracting 7 from 4 for 'avg_pool/AvgPool' (op: 'AvgPool') with input shapes: [32, 4, 4, 2048]"**

Does anyone knows how I can extract features from the conv3_x layers for RPN? How can I implement this successfully? 

Any guidance will be appreciated :)

**Comment by olaff09** [comment]
@thealchemist89,

I see that you're pretty active on this, nice!

Here is my input:

I am guessing you're getting this on train time, the first thing I would think is that the output (X,Y) of your **data_generators.calc_rpn()** function does not mach the size the classifier is expecting after you changed the convolutions. You have to change the **rpn_stride** parameter to match the stride size from conv3_x (I think it is 8?). 

Also, you have to change the output of **get_img_output_length()** function, this one simulates the width, height from the layer you want: You removed one layer from the base, so I would remove the last value from the **filter_sizes** array there, leaving:
`filter_sizes = [7, 3, 1]`

There may be some other details around, but I hope you get the idea. I'd like to test this also so please update me if you're able to get some results.

**Comment by thealchemist89** [comment]
@olaff09 ,

Haha, I see you are getting active on this too, great! Yes, I have changed rpn_stride to 8 to match conv3_x layer.

Additionally, I have also changed pooling_regions = 28, input_shape = (num_rois, 28, 28, 512) under classifer() in resnet.py. All these to match as closely as possible to conv3_x layer outputs.

Currently I am getting "Incompatible shape: [1, 38, 47, 48] vs. [1,75,133,48]"

I will fiddle around with what you said :)

Thanks again!

**Comment by thealchemist89** [comment]
@olaff09,

Thanks mate, it did the trick, i m

**Comment by thealchemist89** [comment]
@olaff09,

Oops, click wrongly. You were right, the filter_sizes have to be adjusted. Instead of [7 3 1 1], it should be [7 3 1].  It looks like it is training now at least albeit very slowly on GTX1080Ti.

**Comment by olaff09** [comment]
Yeah, reducing the stride will create a lot more regions, so computation will increase. About speeding up, I've done some improvements (multi processing) for calc_rpn and non max suppresion, still in early stages but it is working and faster.
